### PR TITLE
feat(FileUploadList): new component

### LINKED ISSUE #337: [Feat]: Add new FileUploadList component
### Component

File Upload List

### Is your feature request related to a problem? Please describe.

Description:
The File Upload List component is designed to facilitate the display and management of files uploaded by users. This component provides a clear and organized list of uploaded files, including essential details such as file name, size, upload date, and available actions (e.g., download, remove, rename, or preview). The component ensures a consistent, accessible, and user-friendly experience aligned with Gene UI design standards.

### Describe the solution you'd like

Requirements:

Component Implementation
Implement the File Upload List component in TypeScript. The component must support dynamic data and handle updates when new files are uploaded or existing ones are removed.
Use SCSS with BEM methodology for styling to maintain a clean separation between structure and presentation.
Integrate design tokens from @geneui/tokens for consistent usage of colors, typography, spacing, borders, and shadows following Gene UI visual guidelines.

**Support Key Features:**

- File Display: Present a simple, organized list showing file name, type icon (based on MIME), size (in human-readable format), and upload date.
- File Actions: Provide per-file actions for Download, Remove, Rename, and Preview (if applicable).
- Dynamic Updates: Automatically reflect changes when users add, remove, or modify files.
- State Management: Display appropriate visual feedback during upload progress, completion, or error states.
- Empty State: Show a dedicated empty state when there are no uploaded files, including a message or upload prompt.
- RTL Support: Ensure full Right-to-Left (RTL) support for layout, alignment, and positioning of controls and action buttons.
- Accessibility: Include full accessibility support with ARIA roles, keyboard navigation, and focus management for all interactive elements.

**Unit Tests:**
Write comprehensive unit tests using Jest and React Testing Library.
Ensure test coverage for all props and states, including edge cases:

- Verify correct rendering of file items with all metadata displayed.
- Test all file actions (Download, Remove, Rename, Preview) for correct event triggering.
- Validate UI updates when files are added or removed dynamically.
- Check display of upload progress and error states.
- Verify Empty State behavior when no files exist.
- Confirm proper RTL rendering and alignment.
- Test keyboard accessibility and focus behavior for all actions.
- Edge cases: empty file array, invalid or missing file details, failed action callback.

**Storybook Stories:**
Create detailed Storybook stories for the File Upload List component.
Showcase all key use cases and states, including:

- Default list view with multiple uploaded files.
- Empty state (no uploaded files).
- Upload in progress (progress indicator example).
- Error state (failed upload or failed removal).
- RTL layout example.
- Interactive controls to add, remove, or rename files dynamically.

Each story should include interactive props controls allowing users to modify the list data and test component behavior.
Document all props, event handlers, and data structure formats.

**Checklist:**

- Component implemented in TypeScript.
- Styling follows SCSS using BEM methodology.
- Design tokens from @geneui/tokens are consistently applied.
- Supports all defined features: file display, actions, dynamic updates, empty and error states.
- Fully accessible (ARIA roles, keyboard navigation, focus management).
- RTL layout supported and verified.
- Unit tests written for all props, interactions, and edge cases.
- Storybook stories created for all component states with interactive examples.
- Implementation visually matches Figma design specifications.

### Describe alternatives you've considered

_No response_

### Are you able to assist to bring the feature to reality?

yes

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by NairaMisakyan**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=1529 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-iaiecxkxwk.chromatic.com/

### CODE REVIEW

**Review by alberthovhannisyan1 — DISMISSED** [review]

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+// Hooks
+import useEllipsisDetection from "@hooks/useEllipsisDetection";
+
+// Styles
+import "./FileUploadList.scss";
+
+/** File type used for row visual state (background and icon color). */
+export type FileType = "visual" | "video" | "audio" | "document";
+
+const VALID_FILE_TYPES: FileType[] = ["visual", "video", "audio", "document"];
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Why "visual" but no image ?

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+function getFileType(type?: string): FileType {
+    if (type && VALID_FILE_TYPES.includes(type as FileType)) {
+        return type as FileType;
+    }
+    return "document";
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: In my opinion it's better to return "unknown" and handle that case (instead of "document")

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+     */
+    uploadingText?: string;
+    /**
+     * Accessible label for the list item (e.g. for screen readers).
+     */
+    "aria-label"?: string;
+}
+
+const FileUploadItem: FC<IFileUploadItem> = (props) => {
+    const {
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Let's move destructuring on line 129

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+     */
+    helperText?: string;
+    /**
+     * Text displayed during upload progress.
+     */
+    uploadingText?: string;
+    /**
+     * Accessible label for the list item (e.g. for screen readers).
+     */
+    "aria-label"?: string;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Why not ariaLabel ?

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+        loading,
+        progressPercent,
+        id,
+        status = "rest",
+        helperText,
+        uploadingText,
+        "aria-label": ariaLabel
+    } = props;
+
+    const hasActions = Array.isArray(actions) && actions.length > 0;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: `const hasActions = actions?.length`

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+    const fileType = getFileType(type);
+    const Icon = icons[fileType];
+    const hasName = name !== undefined && name !== null && name !== "";
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: This constant is unnecessary

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+    const fileName = name ?? "Unnamed file";
+    const hasTime = time != null && time !== "";
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: This constant isn't necessary

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+    const fileTime = hasTime ? time : "--:--";
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: instead of hasTime use time ?? "--:--"

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+    const fileSize = size ?? "Unknown";
+
+    const actionsWithIds = useMemo(() => {
+        if (!actions) return [];
+        return actions.map((action, index) => {
+            const { onCancel, ...restAction } = action;
+            const stableId = action.id || `fileUpload-action-${id}-${index}`;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Why here need index ? And is it really stable ?

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+                            as="span"
+                            variant="labelMediumMedium"
+                        >
+                            {fileName}
+                        </Text>
+                    </Tooltip>
+                </div>
+                {!showProgressLayout && (
+                    <>
+                        {hasTime && (
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: use fileTime here

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+                            <div className="fileUploadList__cell">
+                                <Tooltip text={fileTime} isVisible={isTimeTruncated}>
+                                    <Text ref={timeRef} className="ellipsis-text" as="span" variant="labelMediumMedium">
+                                        {fileTime}
+                                    </Text>
+                                </Tooltip>
+                            </div>
+                        )}
+                        {hasName && (
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Use fileSize for checking

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+                        <ButtonGroup size="small">
+                            {actionsWithIds.map((action) =>
+                                action.Icon ? (
+                                    <Button
+                                        key={action.id}
+                                        {...action}
+                                        layout="text"
+                                        appearance="secondary"
+                                        className="fileUploadList__button"
+                                        disabled={loading && !action.onCancel}
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Let's pass the disabled prop from actions (basically the user should give the disabled prop for each action manually)

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+                                        aria-label={
+                                            action["aria-label"] ?? (action.onCancel ? "Cancel upload" : "File action")
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Same here

File: src/components/molecules/FileUploadList/FileUploadItem.tsx (hunk 4 above)
Comment (NairaMisakyan) [reply src/components/molecules/FileUploadList/FileUploadItem.tsx]: This is project friendly model.

File: src/components/molecules/FileUploadList/FileUploadItem.tsx (hunk 6 above)
Comment (NairaMisakyan) [reply src/components/molecules/FileUploadList/FileUploadItem.tsx]: In case I dont pass name no need to create div, Thats why we need to check if we have value then div can display

File: src/components/molecules/FileUploadList/FileUploadItem.tsx (hunk 7 above)
Comment (NairaMisakyan) [reply src/components/molecules/FileUploadList/FileUploadItem.tsx]: In case I dont pass time no need to create div, Thats why we need to check if we have value then div can display

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+export type FileType = "image" | "video" | "audio" | "file";
+
+const VALID_FILE_TYPES: FileType[] = ["image", "video", "audio", "file"];
+
+/** Returns the given type if it's a valid FileType, otherwise "unknown". */
+function getFileType(type?: string): FileType {
+    if (type && VALID_FILE_TYPES.includes(type as FileType)) {
+        return type as FileType;
+    }
+    return "file";
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Why "file" not unknown ?

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+// Styles
+import "./FileUploadList.scss";
+
+/** File type used for row image state (background and icon color). */
+export type FileType = "image" | "video" | "audio" | "file";
+
+const VALID_FILE_TYPES: FileType[] = ["image", "video", "audio", "file"];
+
+/** Returns the given type if it's a valid FileType, otherwise "unknown". */
+function getFileType(type?: string): FileType {
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Use arrow functions instead

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+    video: PlaySquare,
+    audio: NoteMusical,
+    file: Document
+} as const;
+
+export interface IFileUploadActionProps {
+    /**
+     * The `Icon` component to display in the action button. If not provided, the action button will not be rendered.
+     */
+    Icon?: FC<IconProps>;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Why Icon is not required ?

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 17]
```diff
@@ -0,0 +1,234 @@
...
+     */
+    id?: string;
+    /**
+     * Specifies the name of the button.
+     */
+    name?: string;
+    /**
+     * An ARIA label for the button.
+     */
+    "aria-label"?: string;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Let's use ariaLabel instead so we will keep consistency

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 18]
```diff
@@ -0,0 +1,234 @@
...
+    const timeRef = useRef<HTMLSpanElement>(null);
+    const sizeRef = useRef<HTMLSpanElement>(null);
+    const isNameTruncated = useEllipsisDetection(nameRef, [name]);
+    const isTimeTruncated = useEllipsisDetection(timeRef, [time]);
+    const isSizeTruncated = useEllipsisDetection(sizeRef, [size]);
+
+    const showProgressLayout = loading || status === "error" || status === "warning";
+
+    return (
+        <div className="fileUploadList__itemWrapper">
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: ClassNames are not correct, better to use fileUploadItem / fileUploadListItem instead (for all classes)

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 19]
```diff
@@ -0,0 +1,234 @@
...
+                            as="span"
+                            variant="labelMediumMedium"
+                        >
+                            {name}
+                        </Text>
+                    </Tooltip>
+                </div>
+                {!showProgressLayout && (
+                    <>
+                        {time && (
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: To not repeat same JSX for time and size, lets create mapping logic

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 20]
```diff
@@ -0,0 +1,234 @@
...
+                                </Tooltip>
+                            </div>
+                        )}
+                    </>
+                )}
+                <div className={classNames("fileUploadList__cell", { showProgressLayout })}>
+                    {!!actions?.length && (
+                        <ButtonGroup size="small">
+                            {actionsWithIds.map((action) =>
+                                action.Icon ? (
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: If we are making Icon required this ternary operator can be removed

File: src/components/molecules/FileUploadList/FileUploadList.stories.tsx [hunk 21]
```diff
@@ -0,0 +1,129 @@
...
+    }
+};
+
+const mockData: IFileUploadItem[] = [
+    {
+        name: "Brand-styleguide.pdf",
+        time: "08:05AM",
+        size: "6MB",
+        type: "file",
+        actions: [{ Icon: Eye }, { Icon: Download }, { Icon: RecycleBin }]
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadList.stories.tsx]: To not repeat same actions again and again create const and use the constant instead

File: src/components/molecules/FileUploadList/FileUploadList.stories.tsx [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+        children: args({ control: "false", ...propCategory.content })
+    },
+    args: {},
+    render: (props) => (
+        <FileUploadList {...props}>
+            {mockData.map((item, index) => (
+                <FileUploadItem
+                    key={item.id ?? `fallback-${index}`}
+                    {...item}
+                    ariaLabel={
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadList.stories.tsx]: Why we need this on Storybook ?

File: src/components/molecules/FileUploadList/FileUploadList.stories.tsx [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+        id: args({ control: "false", ...propCategory.content }),
+        name: args({ control: "text", ...propCategory.content }),
+        time: args({ control: "text", ...propCategory.content }),
+        size: args({
+            control: "text",
+            ...propCategory.appearance
+        }),
+        type: args({
+            control: "select",
+            options: ["image", "video", "audio", "file"],
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadList.stories.tsx:46]: Why we are hardcoding the options ? Why it's not taking the possible values from component interface ?

File: src/components/molecules/FileUploadList/FileUploadList.stories.tsx [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+            ...propCategory.appearance
+        }),
+        actions: args({ control: "object", ...propCategory.content }),
+        loading: args({ control: "boolean", ...propCategory.states }),
+        progressPercent: args({ control: "number", ...propCategory.content }),
+        status: args({ control: "select", ...propCategory.states, options: ["rest", "warning", "error"] }),
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadList.stories.tsx]: Why we are hardcoding the options ? Why it's not taking the possible values from component interface ?

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 25]
```diff
@@ -0,0 +1,234 @@
...
+                    type="determinate"
+                    status={status}
+                    helperText={helperText}
+                />
+            )}
+        </div>
+    );
+};
+
+FileUploadItem.displayName = "FileUploadItem";
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: I think we don't need this

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 26]
```diff
@@ -0,0 +1,234 @@
...
+     */
+    onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
+    /**
+     * HTML id attribute for the button element.
+     */
+    id?: string;
+    /**
+     * Specifies the name of the button.
+     */
+    name?: string;
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Action name should be required cause on ButtonGroup when the actions are getting grouped they appear inside Menu and there is empty space when name is blank

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 27]
```diff
@@ -0,0 +1,234 @@
...
+        return actions.map((action) => ({
+            ...action,
+            id: action.id || nanoid()
+        }));
+    }, [actions]);
+
+    const nameRef = useRef<HTMLSpanElement>(null);
+    const timeRef = useRef<HTMLSpanElement>(null);
+    const sizeRef = useRef<HTMLSpanElement>(null);
+    const isNameTruncated = useEllipsisDetection(nameRef, [name]);
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: For useEllipsisDetection the dependencies array is not necessary for FileUploadItem

File: src/components/molecules/FileUploadList/FileUploadItem.tsx [hunk 28]
```diff
@@ -0,0 +1,234 @@
...
+                                <Tooltip text={size} isVisible={isSizeTruncated}>
+                                    <Text ref={sizeRef} className="ellipsis-text" as="span" variant="labelMediumMedium">
+                                        {size}
+                                    </Text>
+                                </Tooltip>
+                            </div>
+                        )}
+                    </>
+                )}
+                <div className={classNames("fileUploadList__cell", { showProgressLayout })}>
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/FileUploadItem.tsx]: Remove { showProgressLayout }

**Review by alberthovhannisyan1 — COMMENTED** [review]
Please check comments

File: src/components/molecules/FileUploadList/FileUploadItem.tsx (hunk 14 above)
Comment (NairaMisakyan) [reply src/components/molecules/FileUploadList/FileUploadItem.tsx]: This is discussion result with team.

File: src/components/molecules/FileUploadList/index.tsx [hunk 29]
```diff
@@ -0,0 +1,2 @@
+export { IFileUploadListProps, default as FileUploadList, default } from "./FileUploadList";
```
Comment (alberthovhannisyan1) [inline src/components/molecules/FileUploadList/index.tsx]: We don't need both default export and and `default as smth` export

**Review by GaroGabrielyan — APPROVED** [review]
